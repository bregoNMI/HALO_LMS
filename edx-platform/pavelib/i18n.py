"""
Internationalization tasks
"""

import subprocess
import sys

from paver.easy import cmdopts, needs, sh, task

from .utils.envs import Env
from .utils.timer import timed

try:
    from pygments.console import colorize
except ImportError:
    colorize = lambda color, text: text

DEFAULT_SETTINGS = Env.DEVSTACK_SETTINGS


@task
@needs(
    "pavelib.prereqs.install_prereqs",
    "pavelib.i18n.i18n_validate_gettext",
)
@cmdopts([
    ("verbose", "v", "Sets 'verbose' to True"),
])
@timed
def i18n_extract(options):
    """
    Extract localizable strings from sources
    """
    verbose = getattr(options, "verbose", None)
    cmd = "i18n_tool extract"

    if verbose:
        cmd += " -v"

    sh(cmd)


@task
@needs("pavelib.i18n.i18n_extract")
@timed
def i18n_generate():
    """
    Compile localizable strings from sources, extracting strings first.
    """
    sh("i18n_tool generate")


@task
@needs("pavelib.i18n.i18n_extract")
@timed
def i18n_generate_strict():
    """
    Compile localizable strings from sources, extracting strings first.
    Complains if files are missing.
    """
    sh("i18n_tool generate --strict")


@task
@needs("pavelib.i18n.i18n_extract")
@timed
def i18n_dummy():
    """
    Simulate international translation by generating dummy strings
    corresponding to source strings.
    """
    sh("i18n_tool dummy")
    # Need to then compile the new dummy strings
    sh("i18n_tool generate")


@task
@timed
def i18n_validate_gettext():
    """
    Make sure GNU gettext utilities are available
    """

    returncode = subprocess.call(['which', 'xgettext'])

    if returncode != 0:
        msg = colorize(
            'red',
            "Cannot locate GNU gettext utilities, which are "
            "required by django for internationalization.\n (see "
            "https://docs.djangoproject.com/en/dev/topics/i18n/"
            "translation/#message-files)\nTry downloading them from "
            "http://www.gnu.org/software/gettext/ \n"
        )

        sys.stderr.write(msg)
        sys.exit(1)


@task
@timed
def i18n_clean():
    """
    Clean the i18n directory of artifacts
    """
    sh('git clean -fdX conf/locale')
