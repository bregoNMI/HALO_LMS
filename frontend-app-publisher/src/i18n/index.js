// Placeholder messages index to be overridden by `make pull_translations`

import { messages as paragonMessages } from '@edx/paragon';
import { messages as footerMessages } from '@edx/frontend-component-footer';

export default [
  footerMessages,
  paragonMessages,
];
