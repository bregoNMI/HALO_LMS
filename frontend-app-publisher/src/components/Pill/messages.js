import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  statusArchived: {
    id: 'publisher.status.archived',
    defaultMessage: 'Archived',
    description: 'Label on a badge element to show status.',
  },
  statusUnsubmitted: {
    id: 'publisher.status.unsubmitted',
    defaultMessage: 'Unsubmitted',
    description: 'Label on a badge element to show status.',
  },
  statusScheduled: {
    id: 'publisher.status.Scheduled',
    defaultMessage: 'Scheduled',
    description: 'Label on a badge element to show status.',
  },
  statusInReview: {
    id: 'publisher.status.inReview',
    defaultMessage: 'In review',
    description: 'Label on a badge element to show status.',
  },
  statusPublished: {
    id: 'publisher.status.published',
    defaultMessage: 'Published',
    description: 'Label on a badge element to show status.',
  },
});

export default messages;
