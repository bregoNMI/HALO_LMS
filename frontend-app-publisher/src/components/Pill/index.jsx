import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { useIntl } from '@edx/frontend-platform/i18n';

import messages from './messages';

import {
  REVIEW_BY_LEGAL,
  REVIEW_BY_INTERNAL,
  PUBLISHED,
  REVIEWED,
  UNPUBLISHED,
  ARCHIVED,
} from '../../data/constants';

const Pill = ({ statuses }) => {
  const { formatMessage } = useIntl();

  const pills = [];
  statuses.forEach((status) => {
    switch (status) {
      case ARCHIVED:
        pills.push({
          text: formatMessage(messages.statusArchived),
          className: 'badge badge-secondary',
        });
        break;
      case UNPUBLISHED:
        pills.push({
          text: formatMessage(messages.statusUnsubmitted),
          className: 'badge badge-warning',
        });
        break;
      case REVIEWED:
        pills.push({
          text: formatMessage(messages.statusScheduled),
          className: 'badge badge-primary',
        });
        break;
      case REVIEW_BY_LEGAL:
      case REVIEW_BY_INTERNAL:
        pills.push({
          text: formatMessage(messages.statusInReview),
          className: 'badge badge-light',
        });
        pills.push({
          text: <i className="fa fa-lock" />,
        });
        break;
      case PUBLISHED:
        pills.push({
          text: formatMessage(messages.statusPublished),
          className: 'badge badge-success',
        });
        break;
      default:
        break;
    }
  });
  return pills.map(pill => (
    <span className={classNames('ml-2', pill.className)} key={pill.text}>
      {pill.text}
    </span>
  ));
};

Pill.propTypes = {
  statuses: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default Pill;
