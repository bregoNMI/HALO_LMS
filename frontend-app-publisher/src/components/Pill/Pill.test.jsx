import React from 'react';
import { render } from 'enzyme';
import { renderToJson } from 'enzyme-to-json';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import Pill from './index';

const PillWrapper = (props) => (
  <IntlProvider locale="en">
    <Pill {...props} />
  </IntlProvider>
);

describe('Pill', () => {
  it('renders without any statuses', () => {
    const component = render(<PillWrapper statuses={[]} />);
    expect(renderToJson(component)).toMatchSnapshot();
  });

  it('renders with invalid statuses', () => {
    const statuses = [undefined, null, 'fnjdsahf'];
    const component = render(<PillWrapper statuses={statuses} />);
    expect(renderToJson(component)).toMatchSnapshot();
  });

  it('renders with a single status', () => {
    const statuses = ['review_by_legal'];
    const component = render(<PillWrapper statuses={statuses} />);
    expect(renderToJson(component)).toMatchSnapshot();
  });

  it('renders with multiple statuses', () => {
    const statuses = ['review_by_internal', 'published'];
    const component = render(<PillWrapper statuses={statuses} />);
    expect(renderToJson(component)).toMatchSnapshot();
  });
});
