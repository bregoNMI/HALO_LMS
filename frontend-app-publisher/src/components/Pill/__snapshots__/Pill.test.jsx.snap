// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill renders with a single status 1`] = `
Array [
  <span
    class="ml-2 badge badge-light"
  >
    In review
  </span>,
  <span
    class="ml-2"
  >
    <i
      class="fa fa-lock"
    />
  </span>,
]
`;

exports[`Pill renders with invalid statuses 1`] = `null`;

exports[`Pill renders with multiple statuses 1`] = `
Array [
  <span
    class="ml-2 badge badge-light"
  >
    In review
  </span>,
  <span
    class="ml-2"
  >
    <i
      class="fa fa-lock"
    />
  </span>,
  <span
    class="ml-2 badge badge-success"
  >
    Published
  </span>,
]
`;

exports[`Pill renders without any statuses 1`] = `null`;
